civic_sandbox/helpers.py

```diff
@@ -0,0 +1,70 @@
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from .models import BikeParking
+from .serializers import BikeParkingSerializer
+import json 
+from django.contrib.gis.geos import GEOSGeometry, MultiPoint, MultiPolygon, MultiLineString
+from pprint import pprint 
+
+
+def sandbox_view_factory(model_class, serializer_class, multi_geom_class, geom_field, attributes, dates):
+    @api_view(['GET'])
+    def sandbox_function(request, date_filter=dates['default_date_filter'], format=None): 
+        print('\n====================================================================================\n')
+        ## get data ##
+        try:
+            if not dates['date_attribute']: 
+                dataset= model_class.objects.all()
+            else:
```

Comment:
I'm not 100% on the details here, but if you are filtering a date column by year, month etc.  you can do something like linked below.

Also worth noting that `result = model.objects.all()` doesn't actually hit the database until you are processing (i.e. looping over, printing, etc.) `result`.  So you can chain filters as necessary. 

https://docs.djangoproject.com/en/2.0/topics/db/queries/#retrieving-specific-objects-with-filters